### PR TITLE
fix(iam): support NotActions/NotResources (#964)

### PR DESCRIPTION
**Pull request by statik**
This adds support for NotActions and NotResources in PolicyDocument.

I updated unit tests as well and did my best to follow CONTRIBUTING.

----

Please read the [contribution guidelines](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-iam/lib/policy-statement.ts [hunk 1]
```diff
@@ -229,6 +243,13 @@ export interface PolicyStatementProps {
    */
   readonly actions?: string[];
 
+  /**
+   * List of not actions to add to the statement
+   *
+   * @default - no actions
+   */
+  readonly notactions?: string[];
```
Comment (jogold) [inline packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: Shouldn't this be called `notActions`? Camel case?

File: packages/@aws-cdk/aws-iam/lib/policy-statement.ts [hunk 2]
```diff
@@ -239,10 +260,17 @@ export interface PolicyStatementProps {
...
+   * @default - no resources
    */
   readonly resources?: string[];
 
+  /**
+   * NotResource ARNs to add to the statement
+   *
+   * @default - no resources
+   */
+  readonly notresources?: string[];
```
Comment (jogold) [inline packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: Shouldn't this be called `notResources`? Camel case?

File: packages/@aws-cdk/aws-iam/lib/policy-statement.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: You are right, it should have been.

File: packages/@aws-cdk/aws-iam/lib/policy-statement.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: You are right, it should have been.

Comment (jogold) [reply packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: You want to change this before v1.5.0?

**Comment by rix0rrr** [comment]
@NGL321 this has been merged prematurely. It does not conform to style standards, and I would say it actually also needs a check against combining `actions` and `notActions` together (which is not allowed according to [the IAM policy grammar](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_grammar.html)).

File: packages/@aws-cdk/aws-iam/lib/policy-statement.ts (hunk 2 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: Hell yes :)

Comment (jogold) [reply packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: I can quickly open a PR to fix the casing, The check against combining `actions` and `notActions` can wait after v1.5.0 I presume.

Comment (jogold) [reply packages/@aws-cdk/aws-iam/lib/policy-statement.ts]: OK you did it already :)

**Pull request merged**